sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -881,7 +881,7 @@ def test_convergence_warnings():
 
     # check that the model fails to converge
     with pytest.warns(ConvergenceWarning):
-        MultiTaskElasticNet(max_iter=1, tol=0).fit(X, y)
+        MultiTaskElasticNet(alpha=0.001, max_iter=1, tol=0).fit(X, y)
```

Comment:
why is changing alpha necessary here ? It converges with only 1 iteration with default alpha ?

Reply:
the default alpha is superior to alpha_max for this pair (X, y). So a duality gap of 0 (up to numerical errors) is reached in 0 iterations.

Reply:
I must be misunderstanding something but the test does not fail on master. How can it fail with this PR ?

Reply:
It is tricky.
The test should fail on master. It passes only because of numerical errors (a duality gap of 1e-13 instead of 0 is returned).
I believe it fails on the PR because a duality gap of 0 is returned (which is the correct value)

Run:
```
import numpy as np
from sklearn.linear_model import MultiTaskElasticNet

random_state = np.random.RandomState(0)
X = random_state.standard_normal((1000, 500))
y = random_state.standard_normal((1000, 3))

clf = MultiTaskElasticNet(max_iter=1, tol=0).fit(X, y)
print(clf.dual_gap_)  # should be 0, is 1e-13
print(np.linalg.norm(clf.coef_))   # we have converged. The initialization is already the solution bc alpha >= alpha_max
```

Reply:
Thanks for the clarification. `tol=0` is rarely a good idea because it's subject to numerical precision issues. We can change the tol or the alpha. But in either case could you add a comment to explain that ?

Reply:
IMO It is more logical to use `tol=-1` (eg, the value of alpha we have used could still be >= alpha_max (it is not))

Where should I put the comment?

Reply:
just above this line, when creating the model instance.